src/dataflow/src/source/kafka.rs

```diff
@@ -656,172 +810,220 @@ where
+                        if bytes_read > 0 {
+                            KAFKA_BYTES_READ_COUNTER.inc_by(bytes_read);
+                        }
                         activator.activate();
                         return SourceStatus::Alive;
                     }
-                    Some(ts) => {
-                        // Note: empty and null payload/keys are currently
-                        // treated as the same thing.
-                        let key = message.key.unwrap_or_default();
-                        let out = message.payload.unwrap_or_default();
-                        // Entry for partition_metadata is guaranteed to exist as messages
-                        // are only processed after we have updated the partition_metadata for a
-                        // partition and created a partition queue for it.
-                        cp_info
-                            .partition_metadata
-                            .get_mut(&partition)
-                            .unwrap()
-                            .offset = offset;
-                        bytes_read += key.len() as i64;
-                        bytes_read += out.len() as i64;
-                        let ts_cap = cap.delayed(&ts);
-                        output.session(&ts_cap).give(SourceOutput::new(
-                            key,
-                            out,
-                            Some(Into::<KafkaOffset>::into(offset).offset),
-                        ));
-
-                        partition_metrics.offset_ingested.set(offset.offset);
-                        partition_metrics.messages_ingested.inc();
-
-                        downgrade_capability(
-                            &id,
-                            cap,
-                            &mut cp_info,
-                            &mut dp_info,
-                            &timestamp_histories,
-                        );
-                    }
                 }
-
-                if timer.elapsed().as_millis() > 10 {
-                    // We didn't drain the entire queue, so indicate that we
-                    // should run again. We suppress the activation when the
-                    // queue is drained, as in that case librdkafka is
-                    // configured to unpark our thread when a new message
-                    // arrives.
-                    if bytes_read > 0 {
-                        KAFKA_BYTES_READ_COUNTER.inc_by(bytes_read);
-                    }
-                    activator.activate();
-                    return SourceStatus::Alive;
+                if bytes_read > 0 {
+                    KAFKA_BYTES_READ_COUNTER.inc_by(bytes_read);
                 }
+                // Ensure that we activate the source frequently enough to keep downgrading
+                // capabilities, even when no data has arrived
+                // TODO(ncrooks): make this configurable
+                activator.activate_after(Duration::from_millis(1));
```

Comment:
The timeout looks like a mistake

Reply:
It should have been Duration::from_secs, yes.